python-modules/cis_crypto/cis_crypto/operation.py

```diff
@@ -73,20 +82,63 @@ def _get_public_key(self):
+            key_name = self.config('public_key_name', namespace='cis', default='access-file-key')
             file_name = '{}.pub.pem'.format(key_name)
             fh = open((os.path.join(key_dir, file_name)), 'rb')
             key_content = fh.read()
             key_construct = jwk.construct(key_content, 'RS256')
             return key_construct.to_dict()
-        elif self.well_known_mode is 'http':
+        elif self.well_known_mode == 'http' or self.well_known_mode == 'https':
             # Go get it from the .well-known endpoint and load as json
             # return a dictionary of the json loaded data
-            pass
+            well_known_url = self.config('well_known_url', namespace='cis')
+            res = requests.get(well_known_url)
+
+            if res.status_code == 200:
+                key_list = self._reduce_keys(res.json())
+                return key_list
+
+    def _reduce_keys(self, well_known_response):
+        access_file_keys = well_known_response['access_file']['jwks_keys']
+        publisher_keys = well_known_response['publishers_supported']
+
+        keys = []
+
+        for key in access_file_keys:
+            count = 0
+            keys.append(
+                {
+                    'access_file_key_{}'.format(count): access_file_keys.get(str(count))
+                }
+            )
+            count = count + 1
+
+        for k in publisher_keys:
+            count = 0
+            keys.append(
+                {
+                    k: publisher_keys[k]
+                }
+            )
+
+        return keys
 
     def jws(self):
         """Assumes you loaded a payload.  Return the same jws or raise a custom exception."""
-        jwk = self._get_public_key()
-        sig = jws.verify(self.jws_signature, jwk, algorithms='RS256', verify=True)
-        # XXX TBD failed verification will raise JWSError log in mozilla-iam format the custom exception
-        return sig
+        key_material = self._get_public_key()
+        if isinstance(key_material, dict):
+            sig = jws.verify(self.jws_signature, key_material, algorithms='RS256', verify=True)
+
+        if isinstance(key_material, list):
+            logger.debug('Multiple keys returned.  Attempting match.')
+            for key in key_material:
+                for k_id in key_material[key_material.index(key)]:
+                    logger.info('Attempting to match against: {}'.format(k_id))
+                    key_content = key_material[key_material.index(key)][k_id]
+                    try:
+                        sig = jws.verify(self.jws_signature, key_content, algorithms='RS256', verify=True)
+                        logger.info('Matched a verified signature for: {}'.format(k_id))
+                        return sig
+                    except JWSError as e:
+                        logger.error(e)
+        raise JWSError('The signature could not be verified for any trusted key.')
```

Comment:
Maybe instead set `sig` to `False` and have your upstream code check if `sig` is not False (instead of raising an exception), just an idea, this way looks fine too.